pymc3/gp/mean.py

```diff
@@ -1,12 +1,61 @@
+import numpy as np
 import theano.tensor as tt
 
-__all__ = ['Zero', 'Constant']
+__all__ = ['Zero', 'Constant', 'Linear']
 
-class Mean(object):
+
+class ParameterizedFunction(object):
+    R"""
+    Base class for all functions.  The input_dim and active_dims parameters
+    allow for combining functions which act on subspaces to be added and
+    multiplied using Python built-in operators like "+" and "*".
+
+    Parameters
+    ----------
+    input_dim : integer
+        The number of input dimensions, or columns of X (or Z) the kernel will operate on.
+    active_dims : A list of booleans whose length equals input_dim.
+	The booleans indicate whether or not the covariance function operates
+        over that dimension or column of X.
+    """
+
+    def __init__(self, input_dim, active_dims=None):
```

Comment:
hmm I'm not sure.  It's certainly the most likely thing a user would do.  When these functions are instantiated they don't know about `X` yet, so if a user forgets to set it explicitly they'll get an error at runtime.